test/libsolidity/semanticTests/interfaceID/interfaces.sol

```diff
@@ -0,0 +1,67 @@
+interface HelloWorld {
+    function hello() external pure;
+    function world(int) external pure;
+}
+
+interface HelloWorldDerived is HelloWorld {
+    function other() external pure;
+}
+
+interface ERC165 {
+    /// @notice Query if a contract implements an interface
+    /// @param interfaceID The interface identifier, as specified in ERC-165
+    /// @dev Interface identification is specified in ERC-165. This function
+    ///  uses less than 30,000 gas.
+    /// @return `true` if the contract implements `interfaceID` and
+    ///  `interfaceID` is not 0xffffffff, `false` otherwise
+    function supportsInterface(bytes4 interfaceID) external view returns (bool);
+}
+
+contract Test {
+    bytes4 public ghello_world_interfaceId = type(HelloWorld).interfaceId;
+    bytes4 public ERC165_interfaceId = type(ERC165).interfaceId;
+
+    function hello() public pure returns (bytes4 data){
+        HelloWorld i;
+        return i.hello.selector;
+    }
+
+    function world() public pure returns (bytes4 data){
+        HelloWorld i;
+        return i.world.selector;
+    }
+
+    function hello_world() public pure returns (bytes4 data){
+        // HelloWorld i;
+        // return i.hello.selector ^ i.world.selector; // = 0xc6be8b58
+        return 0xc6be8b58;
```

Comment:
I return here `0xc6be8b58`, because `xor` is not yet supported in yul.

Reply:
Indeed! Let me create a PR to change that...

Reply:
https://github.com/ethereum/solidity/pull/8679